ibc-testkit/src/testapp/ibc/core/types.rs

```diff
@@ -39,69 +48,106 @@ use crate::testapp::ibc::clients::mock::client_state::{
+use crate::testapp::ibc::utils::blocks_since;
 pub const DEFAULT_BLOCK_TIME_SECS: u64 = 3;
 
 /// An object that stores all IBC related data.
-#[derive(Clone, Debug, Default)]
-pub struct MockIbcStore {
-    /// The set of all clients, indexed by their id.
-    pub clients: BTreeMap<ClientId, MockClientRecord>,
-
-    /// Tracks the processed time for clients header updates
-    pub client_processed_times: BTreeMap<(ClientId, Height), Timestamp>,
-
-    /// Tracks the processed height for the clients
-    pub client_processed_heights: BTreeMap<(ClientId, Height), Height>,
-
-    /// Counter for the client identifiers, necessary for `increase_client_counter` and the
-    /// `client_counter` methods.
-    pub client_ids_counter: u64,
-
-    /// Association between client ids and connection ids.
-    pub client_connections: BTreeMap<ClientId, ConnectionId>,
-
-    /// All the connections in the store.
-    pub connections: BTreeMap<ConnectionId, ConnectionEnd>,
-
-    /// Counter for connection identifiers (see `increase_connection_counter`).
-    pub connection_ids_counter: u64,
-
-    /// Association between connection ids and channel ids.
-    pub connection_channels: BTreeMap<ConnectionId, Vec<(PortId, ChannelId)>>,
-
-    /// Counter for channel identifiers (see `increase_channel_counter`).
-    pub channel_ids_counter: u64,
-
-    /// All the channels in the store. TODO Make new key PortId X ChannelId
-    pub channels: PortChannelIdMap<ChannelEnd>,
-
-    /// Tracks the sequence number for the next packet to be sent.
-    pub next_sequence_send: PortChannelIdMap<Sequence>,
-
-    /// Tracks the sequence number for the next packet to be received.
-    pub next_sequence_recv: PortChannelIdMap<Sequence>,
-
-    /// Tracks the sequence number for the next packet to be acknowledged.
-    pub next_sequence_ack: PortChannelIdMap<Sequence>,
-
-    pub packet_acknowledgement: PortChannelIdMap<BTreeMap<Sequence, AcknowledgementCommitment>>,
-
-    /// Constant-size commitments to packets data fields
-    pub packet_commitment: PortChannelIdMap<BTreeMap<Sequence, PacketCommitment>>,
-
-    /// Used by unordered channel
-    pub packet_receipt: PortChannelIdMap<BTreeMap<Sequence, Receipt>>,
+#[derive(Debug)]
+pub struct MockIbcStore<S>
+where
+    S: ProvableStore + Debug,
+{
+    /// Handle to store instance.
+    /// The module is guaranteed exclusive access to all paths in the store key-space.
+    pub store: SharedStore<S>,
+    /// Counter for clients
+    pub client_counter: Arc<Mutex<u64>>,
+    /// Counter for connections
+    pub conn_counter: Arc<Mutex<u64>>,
+    /// Counter for channels
+    pub channel_counter: Arc<Mutex<u64>>,
+    /// Tracks the processed time for client updates
+    pub client_processed_times: Arc<Mutex<BTreeMap<(ClientId, Height), Timestamp>>>,
+    /// Tracks the processed height for client updates
+    pub client_processed_heights: Arc<Mutex<BTreeMap<(ClientId, Height), Height>>>,
```

Comment:
Will the type of these fields be updated once we merge #1069?

Reply:
Yes. and I need to update the corresponding methods in the `*Context` implementations.

Reply:
Can this be handled as part of this PR too?

Reply:
sure! I can take care of it :slightly_smiling_face: